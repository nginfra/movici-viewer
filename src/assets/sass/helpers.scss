.is-size-6-half {
  font-size: 0.85rem !important;
}

.text-ellipsis {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: pre;
}

// .x-overflow is used alongside with FixedPosition.ts as the overflow parent
//
.overflow,
.x-overflow {
  overflow: auto;
  padding-right: 0.5rem;
  margin-right: -0.5rem;
}

.overflow-hidden {
  overflow: hidden !important;
}

.overflow-hover {
  overflow: hidden;
  &:hover {
    overflow: auto;
    padding-right: 0.5rem;
    margin-right: -0.5rem;
  }
}

button {
  &.is-borderless {
    border: 0;
  }
  &.is-transparent {
    background-color: transparent;
    color: $black;
  }
  &.is-round {
    border-radius: 100%;
  }
  &.movici {
    background: $white;
    color: $primary;
    border: 2px solid $white-ter !important;
    &:hover {
      color: $primary;
      background: $primary-light;
      border-color: $grey-light;
    }
    &:active {
      color: $primary;
      border-color: $primary !important;
    }
  }
}

[required] {
  label,
  .label {
    &::after {
      margin-left: 2px;
      content: '*';
      color: $grey;
      font-weight: bold;
    }
  }
}
