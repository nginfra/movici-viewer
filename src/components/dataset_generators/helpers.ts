// @ts-nocheck
import GeoJSON from 'ol/format/GeoJSON';
import { transformGeoJsonToCRS } from '@/crs';
import { MoviciError } from '@/errors';
import { DatasetGeneratorDataset } from '@/types';

export function findDataset(type: string, currentDataset: DatasetGeneratorDataset) {
  let currentChild, result;

  if (type === currentDataset.type) {
    return currentDataset;
  } else {
    if (currentDataset.childs === undefined) {
      return null;
    }

    // Loop over child types
    for (let childTypeIdx = 0; childTypeIdx < currentDataset.childs.length; ++childTypeIdx) {
      const currentChildType = currentDataset.childs[childTypeIdx];

      for (let childIdx = 0; childIdx < currentChildType.datasets.length; ++childIdx) {
        currentChild = currentChildType.datasets[childIdx];
        // Search in the current child
        result = findDataset(type, currentChild);

        if (result !== null) {
          return result;
        }
      }

      // Return the result if the node has been found
    }

    // The node has not been found and we have no more options
    return null;
  }
}

export function validateGeoJSON(textData: string) {
  let feature;
  try {
    feature = new GeoJSON().readFeature(textData);
  } catch (e) {
    console.warn('Invalid GeoJSON supplied');
    failMessage('Invalid GeoJSON supplied');
    return false;
  }

  const featureType = feature?.getGeometry()?.getType();
  if (featureType?.toLowerCase() == 'polygon') {
    return true;
  }

  failMessage(`Feature is not a Polygon but a ${featureType}`);
  return false;
}

export function transformGeoJSON(geojson: GeoJSON) {
  let transformed;
  try {
    transformed = transformGeoJsonToCRS(geojson);
  } catch (e) {
    console.warn(e.message);
    if (e instanceof MoviciError) {
      failMessage(e.message);
    }
  }
  return transformed;
}
