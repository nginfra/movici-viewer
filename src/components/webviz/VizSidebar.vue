<template>
  <div class="sidebar-page">
    <section class="sidebar-layout">
      <BSidebar type="is-light" fullheight open :reduce="reduced" :can-cancel="[]">
        <div class="p-2 pt-3 pb-5">
          <div class="block">
            <template v-if="reduced">
              <div class="has-text-centered">
                <router-link :to="{ name: 'Home' }">
                  <div class="logo-symbol"></div>
                </router-link>
                <BButton
                  ref="reduce-button"
                  @click="toggleReduce"
                  icon-right="angle-right"
                  size="is-small"
                />
              </div>
            </template>
            <template v-else>
              <div>
                <router-link :to="{ name: 'Home' }">
                  <div class="logo"></div>
                </router-link>
              </div>
              <BButton
                ref="reduce-button"
                @click="toggleReduce"
                icon-right="angle-left"
                size="is-small"
              />
            </template>
          </div>
          <slot name="below-header" :reduce="reduced"></slot>
        </div>
        <slot name="content" :reduce="reduced"></slot>
      </BSidebar>
    </section>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component({
  name: 'VizSidebar'
})
export default class VizSidebar extends Vue {
  reduced = false;

  toggleReduce() {
    this.reduced = !this.reduced;
  }
}
</script>
