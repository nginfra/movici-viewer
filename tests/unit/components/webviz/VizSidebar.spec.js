import VizSidebar from '@/components/webviz/VizSidebar';
import { createComponentWrapper } from '../../helpers';

describe('VizSidebar.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = createComponentWrapper(VizSidebar, {}, { shallowMount: true });
  });
  it('renders correctly when expanded', () => {
    wrapper.setData({ reduced: false });
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when reduced', async () => {
    await wrapper.setData({ reduced: true });
    expect(wrapper).toMatchSnapshot();
  });
  it('reduces on click', () => {
    wrapper.setData({ reduced: false });
    wrapper.findComponent({ name: 'BButton' }).vm.$emit('click');
    expect(wrapper.vm.reduced).toBe(true);
  });
  it('expands when reduced', () => {
    wrapper.setData({ reduced: true });
    wrapper.findComponent({ name: 'BButton' }).vm.$emit('click');
    expect(wrapper.vm.reduced).toBe(false);
  });
});
