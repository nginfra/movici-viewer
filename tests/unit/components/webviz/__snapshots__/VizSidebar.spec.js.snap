// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VizSidebar.vue renders correctly when expanded 1`] = `
<div class="sidebar-page">
  <section class="sidebar-layout">
    <bsidebar-stub open="true" type="is-light" position="fixed" fullheight="true" cancancel="" oncancel="[Function]" scroll="clip">
      <div class="p-2 pt-3 pb-5">
        <div class="block">
          <div><a>
              <div class="logo"></div>
            </a></div>
          <bbutton-stub size="is-small" iconright="angle-left" nativetype="button" tag="button"></bbutton-stub>
        </div>
      </div>
    </bsidebar-stub>
  </section>
</div>
`;

exports[`VizSidebar.vue renders correctly when reduced 1`] = `
<div class="sidebar-page">
  <section class="sidebar-layout">
    <bsidebar-stub open="true" type="is-light" position="fixed" fullheight="true" cancancel="" oncancel="[Function]" scroll="clip" reduce="true">
      <div class="p-2 pt-3 pb-5">
        <div class="block">
          <div class="has-text-centered"><a>
              <div class="logo-symbol"></div>
            </a>
            <bbutton-stub size="is-small" iconright="angle-right" nativetype="button" tag="button"></bbutton-stub>
          </div>
        </div>
      </div>
    </bsidebar-stub>
  </section>
</div>
`;
